src/wallet/wallet.cpp

```diff
@@ -3406,6 +3406,14 @@ void CWallet::postInitProcess()
 
 bool CWallet::BackupWallet(const std::string& strDest) const
 {
+    if (m_chain) {
+        CBlockLocator loc;
+        WITH_LOCK(cs_wallet, chain().findBlock(m_last_block_processed, FoundBlock().locator(loc)));
+        if (!loc.IsNull()) {
+            WalletBatch batch(GetDatabase());
+            batch.WriteBestBlock(loc);
+        }
+    }
```

Comment:
In commit "wallet: Write best block to disk before backup" (888a167446901c12960e6776613a3a1fb9789f59)

I wonder if it makes sense to put this in the CWallet::Flush method and call Flush() here. For example, this also seems like a good thing to do when unloading a wallet.

Reply:
I just looked into this and from my understanding `CWallet::Flush()` delegates to the `Flush()` method of the underlying Database and that method is a [no-op in SQLite](https://github.com/bitcoin/bitcoin/blob/master/src/wallet/sqlite.h#L155). Since we are moving to SQLite and remove BDB entirely my guess is that the `Flush()` method will be removed at some point in the future as well.